template.yml.erb

```diff
@@ -41,6 +41,146 @@ Globals:
 Conditions:
   IsDevCondition: !Equals [!Ref BaseDomainName, "dev-code.org"]
 Resources:
+  HttpDomain:
```

Comment:
For future reference, what was your workflow for figuring out what resources you needed for this?

Reply:
Oof. In this particular case, I just copied and pasted everything from the WebSocket API that was domain name related and then didn't really touch it besides updating all the names to use "http." Writing that out makes me realize this may be a good opportunity to collect these into a grouping similar to the authorizer lambda so it's a little clearer that those two sections are repeated code.

Reply:
Actually, were you asking about figuring out the resources I needed for the new domain name or for the entire set of work here?

Reply:
Regarding all the resources needed for the API Gateway api, there were a few steps

- First, Sanchit had created a working API Gateway api. I was able to use this as a reference to start off creating my version.
- Part way through, I realized the type of API he'd created (REST) wasn't supported in CloudFormation. So then I made a new version of his API that used a different type (HTTP). I started using this as a reference. I started off without the authorizer in order to make testing easier (no need to generate a JWT every time)
- When working on the API, I started with a fresh yml template and hacked the deploy.sh script to deploy mine instead of the main template. This allowed me to iterate much faster. (deploy time of ~30s rather than ~5min)
- I also used a .yml template rather than a .yml.erb template. This just generally made things easier since I didn't have to mix syntaxes
- To figure out the resources I needed, I did a combination of referencing the docs (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html) which are about as easy to read as ancient Greek, referencing the sample I'd created, and referencing the WebSocket API resources that Suresh had created. As I got closer to the right solution, the docs did get more useful, but more so in telling me how to assemble things I already had figured out I needed, rather than telling me what I needed.
- To be completely honest, though, there was a lot of guess-and-check. Looking back at my pomodoro history in Marinara, I spent 17 hours over the past two weeks in deep focus just trying various things out in addition to time I spent bouncing ideas off the infra team and time I didn't track. The pomodoros were helpful because I could use them as a litmus test for whether I was making any progress - two pomodoros without any progress was a pretty clear indicator that I was stuck (or really needed a break)

Reply:
Yeah I was looking for a overall summary, just like ^^^, thanks! lol at "ancient Greek"

Reply:
Oh one other item. Only the infra team can update the iam permissions template. I hit a couple snags with this, but ultimately ended up not needing to update it after all. Adding a completely new resource type (such as db tables) will likely need an update to the IAM permissions.